components/homme/src/test_src/dry_planar.F90

```diff
@@ -170,11 +172,141 @@ subroutine planar_rising_bubble_init(elem,hybrid,hvcoord,nets,nete)
+  real(rl):: x,y,hyam,hybm,hyai,hybi           
+  real(rl):: pi(nlevp), dpm(nlev), th0(nlevp), th0m(nlev), ai(nlevp), bi(nlevp), rr
+
+#ifdef MODEL_THETA_L
+
+  if (hybrid%masterthread) then
+     write(iulog,*) 'initializing hot bubble with'
+     print *, 'Lx, Ly =', Lx, Ly
+     print *, 'Sx, Sy =', Sx, Sy
+     print *, 'bubble_T0',  bubble_T0
+     print *, 'bubble_dT', bubble_dT
+     print *, 'bubble_xycenter', bubble_xycenter
+     print *, 'bubble_zcenter', bubble_zcenter
+     print *, 'bubble_ztop', bubble_ztop
+     print *, 'bubble_radius', bubble_radius
+     print *, 'bubble_cosine', bubble_cosine
+  endif
+
+  !evenly spaced with reversed indexing, just like in homme eta coord
+  call get_evenly_spaced_z(zi,zm, 0.0_rl,bubble_ztop)
+
+  ! set initial conditions
+  do ie = nets,nete
+    do j=1,np; do i=1,np
+
+      ! get horizontal coordinates at column i,j
+      x  = elem(ie)%spherep(i,j)%lon
+      y  = elem(ie)%spherep(i,j)%lat
+
+      do k=1,nlevp
+
+        rr = sqrt( (x-bubble_xycenter)   *(x-bubble_xycenter)+&
+                   (y-bubble_xycenter)   *(y-bubble_xycenter)+&
+                   (zi(k)-bubble_zcenter)*(zi(k)-bubble_zcenter) )
+
+        !set pot. temperature on interfaces
+        if ( rr < bubble_radius ) then 
+          if (bubble_cosine) then
+            th0(k) = bubble_T0 + bubble_dT * cos(rr*dd_pi / bubble_radius / 2.d0)
+          else
+            th0(k) = bubble_T0 + bubble_dT
+          endif
+
+        else
+          th0(k) = bubble_T0
+        endif
+
+        !set interface pressure from const lapse rate, does not depend on x,y, or _dT
+        pi(k) =  p0*( (270.0 - zi(k)*g/cp)/270.0  )**(1.0/kappa)
+      enddo
+
+      elem(ie)%state%ps_v(i,j,:) = pi(nlevp)
+      elem(ie)%state%v = 0.0
+      elem(ie)%state%w_i= 0.0
+
+      do k=1,nlev
+        !set pottemp, dp, other state vars on midlevels
+        th0m(k)=(th0(k)+th0(k+1))/2.0
+        dpm(k)=(pi(k+1)-pi(k))
+
+        elem(ie)%state%dp3d(i,j,k,:)   = dpm(k)
+      
+        elem(ie)%state%phis(i,j) = 0.0
+        elem(ie)%state%phinh_i(i,j,k,:) = zi(k)*g
+
+        elem(ie)%state%vtheta_dp(i,j,k,:) = dpm(k)*th0m(k)
+      enddo !k loop
+
+    enddo; enddo !i,j loop
+  enddo !ie loop
+
+  !create hybrid coords now from the 1st column
+  !note that this code should not depend on partitioning
+  !it depends on pressure, whcih is init-ed in the same way for all elements/gll
+  ai(:) = 0.0; bi(:) = 0.0
+  ai(1) = pi(1)/p0
+  bi(nlevp) = 1.0
+
+  do k=2,nlev
+    bi(k) = 1.0 - zi(k)/zi(1)
+
+    !restore ai frop given pressure
+    ai(k)=(pi(k)-bi(k)*pi(nlev))/p0
+  enddo
+
+  hvcoord%hyai = ai
+  hvcoord%hybi = bi
+
+  !set : hyam hybm 
+  hvcoord%hyam = 0.5_rl *(ai(2:nlev+1) + ai(1:nlev))
+  hvcoord%hybm = 0.5_rl *(bi(2:nlev+1) + bi(1:nlev))
+
+  !  call set_layer_locations: sets  etam, etai, dp0, checks that Am=ai/2+ai/2
+  call set_layer_locations(hvcoord, .true., hybrid%masterthread)
+
```

Comment:
This should probably be subroutine, since setting equally spaced z levels for a given p/T profile is going to show up in a lot of the planar test cases.